R/arrange.R

```diff
@@ -210,11 +214,60 @@ locale_to_chr_proxy_collate <- function(locale,
     return(sort_key_generator(locale))
   }
 
-  abort("`.locale` must be a string.", call = error_call)
+  abort("`.locale` must be a string or `NULL`.", call = error_call)
 }
 
 sort_key_generator <- function(locale) {
   function(x) {
     stringi::stri_sort_key(x, locale = locale)
   }
 }
+
+# ------------------------------------------------------------------------------
+
+dplyr_order_legacy <- function(data, direction) {
+  if (ncol(data) == 0L) {
+    # Work around `order(!!!list())` returning `NULL`
+    return(seq_len(nrow(data)))
+  }
+
+  proxies <- map2(data, direction, dplyr_proxy_order_legacy)
+  proxies <- unname(proxies)
+
+  inject(order(!!!proxies, decreasing = FALSE, na.last = TRUE))
```

Comment:
```suggestion
  inject(order(!!!proxies))
```

Seeing the explicit values makes me think you're overriding the defaults